org/mozilla/jss/ssl/javax/JSSSession.java

```diff
@@ -0,0 +1,158 @@
+package org.mozilla.jss.ssl.javax;
+
+import java.security.cert.Certificate;
+import javax.security.cert.X509Certificate;
+import java.security.Principal;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.SSLSessionContext;
+import javax.net.ssl.SSLPeerUnverifiedException;
+
+public class JSSSession implements SSLSession {
+    private int application_buffer_size;
+    private String cipher_suite;
+    private long creation_time;
+    private byte[] session_id;
+    private long last_access_time;
+    private Certificate[] local_certificates;
+    private Principal local_principal;
+    private int packet_buffer_size;
+    private X509Certificate[] peer_chain;
+    private Certificate[] peer_certificates;
+    private String peer_host;
+    private int peer_port;
+    private Principal peer_principal;
+    private String protocol;
+
+    JSSSession(int buffer_size) {
+        creation_time = System.currentTimeMillis();
+        application_buffer_size = buffer_size;
+        packet_buffer_size = buffer_size;
+        setLastAccessedTime();
+    }
+
+    public byte[] getId() {
+        return session_id;
+    }
+
+    protected void setId(byte[] new_id) {
+        session_id = new_id;
+        setLastAccessedTime();
+    }
+
+    public SSLSessionContext getSessionContext() {
+        return null;
+    }
+
+    public long getCreationTime() {
+        return creation_time;
+    }
+
+    public long getLastAccessedTime() {
+        return last_access_time;
+    }
+
+    protected void setLastAccessedTime() {
+        last_access_time = System.currentTimeMillis();
+    }
+
+    public void invalidate() {}
+    public boolean isValid() { return true; }
+    public void putValue(String name, Object value) {}
+    public Object getValue(String name) { return null; }
+    public void removeValue(String name) {}
+    public String[] getValueNames() { return null; }
+
+    public Certificate[] getPeerCertificates() {
+        return peer_certificates;
+    }
+
+    protected void setPeerCertificates(Certificate[] new_certs) {
+        peer_certificates = new_certs;
+        setLastAccessedTime();
```

Comment:
Not sure if this is correct. IIUC the "last accessed time" is used to find abandoned session (i.e. no more traffic). See https://docs.oracle.com/javase/8/docs/api/javax/net/ssl/SSLSession.html#getLastAccessedTime--:
```
Access indicates a new connection being established using session data. Application level operations, 
such as getting or setting a value associated with the session, are not reflected in this access time.
```

Reply:
Yeah, this'll eventually be updated. I want to get these values from NSS directly.